### PR TITLE
Add boskos accounts for EKS build cluster

### PR DESCRIPTION
**Pull request by xmudrii**
As discussed with @hh and @Riaankl privately, I'm adding 10 AWS accounts to be used by boskos in the EKS build cluster. I'll leave some questions and remarks as comments.

Next steps would be:

- review this PR
- someone (e.g. @hh or @ameukam) applies this PR and either
  - give me access to those accounts so that I can create IAM users for boskos
  - create IAM users for boskos and give me credentials for those IAM users

/assign @ameukam @dims @hh @Riaankl

### CODE REVIEW

File: infra/aws/terraform/management-account/organization-accounts-boskos.tf [hunk 1]
```diff
@@ -41,3 +40,167 @@ module "k8s_infra_e2e_boskos_003" {
   iam_user_access_to_billing = "ALLOW"
   parent_id                  = aws_organizations_organizational_unit.boskos.id
 }
+
+##############################################
+# Accounts used by boskos in the EKS cluster #
+##############################################
+
+module "k8s_infra_e2e_boskos_004" {
```
Comment (xmudrii) [inline infra/aws/terraform/management-account/organization-accounts-boskos.tf:48]: I'm not sure:

* what are accounts 001 - 003 used for (I assume for boskos running in GKE cluster, but I don't find any references to those accounts)
* do we want to use different numbering for accounts used in EKS cluster, e.g. 100 - 110

File: infra/aws/terraform/management-account/organization-accounts-boskos.tf [hunk 2]
```diff
@@ -41,3 +40,167 @@ module "k8s_infra_e2e_boskos_003" {
...
+  email                      = "k8s-infra-aws-admins+k8s-infra-e2e-boskos-004@kubernetes.io"
+  iam_user_access_to_billing = "ALLOW"
+  parent_id                  = aws_organizations_organizational_unit.boskos.id
+
+  tags = {
+    "production"  = "true",
+    "environment" = "prod",
+    "group"       = "sig-k8s-infra",
+    "service"     = "boskos"
+  }
```
Comment (xmudrii) [inline infra/aws/terraform/management-account/organization-accounts-boskos.tf:61]: I've seen that we add tags for all new AWS accounts, so I added some tags. I'm not sure if those tags are correct and needed, so please check.

File: infra/aws/terraform/management-account/organization-accounts-boskos.tf (hunk 1 above)
Comment (xmudrii) [reply infra/aws/terraform/management-account/organization-accounts-boskos.tf]: Done.

**Comment by dims** [comment]
/approve
/lgtm

thanks @xmudrii

File: infra/aws/terraform/management-account/organization-accounts-boskos.tf (hunk 2 above)
Comment (ameukam) [reply infra/aws/terraform/management-account/organization-accounts-boskos.tf:61]: We currently only have policies for keys `service` and `group`. The rest of the tags are for the moment optional.

File: infra/aws/terraform/management-account/organization-accounts-boskos.tf [hunk 3]
```diff
@@ -41,3 +40,167 @@ module "k8s_infra_e2e_boskos_003" {
   iam_user_access_to_billing = "ALLOW"
   parent_id                  = aws_organizations_organizational_unit.boskos.id
 }
+
+##############################################
+# Accounts used by boskos in the EKS cluster #
+##############################################
+
+module "k8s_infra_eks_e2e_boskos_001" {
```
[the remark this answers is not in the document]
Comment (xmudrii) [reply infra/aws/terraform/management-account/organization-accounts-boskos.tf:48]: See this comment: https://github.com/kubernetes/k8s.io/pull/5185#discussion_r1179394039

File: infra/aws/terraform/management-account/organization-accounts-boskos.tf (hunk 2 above)
Comment (xmudrii) [reply infra/aws/terraform/management-account/organization-accounts-boskos.tf:61]: @ameukam Should I change something or leave as it is?

Comment (ameukam) [reply infra/aws/terraform/management-account/organization-accounts-boskos.tf:61]: You can leave this as it is. Tags are mutable and can be updated later.

**Comment by hh** [comment]
These accounts are all in the Kubernetes 1Password AWS CI Vault:
image

I've shared 7 day links to them with @xmudrii 

I'd like to possibly create more vaults, but I'm not yet familiar enough with 1Password.

**Pull request merged**